tests/rptest/tests/datalake/query_engine_base.py

```diff
@@ -66,3 +66,11 @@ def max_translated_offset(self, namespace, table, partition) -> int:
         query = f"select max(redpanda.offset) from {namespace}.{self.escape_identifier(table)} where redpanda.partition={partition}"
         with self.run_query(query) as cursor:
             return cursor.fetchone()[0]
+
+    @abstractmethod
+    def count_parquet_files(self, namespace, table) -> int:
+        ...
```

Comment:
What does this look like when not implemented? Is it a no-op? Should we raise an exception like we do elsewhere?

Reply:
You can't actually instantiate a class which inherits from `abc.ABC` and has any `@abstractmethod` members. The body is dead/unreachable code.

```
>>> from abc import ABC, abstractmethod
>>> class A(ABC):
...     @abstractmethod
...     def a(self):
...         print("foo")
...
>>> A().a()
Traceback (most recent call last):
  File "<python-input-13>", line 1, in <module>
    A().a()
    ~^^
TypeError: Can't instantiate abstract class A without an implementation for abstract method 'a'
>>>
```